app/controllers/posts_controller.rb

```diff
@@ -1,15 +1,14 @@
 class PostsController < ApplicationController
   # GET /user/:user_id/posts
   def index
-    @user = User.find(params[:user_id])
-    @user_posts = Post.where(author_id: params[:user_id]).all
+    @user = User.includes(:posts).find(params[:user_id])
```

Comment:
You are not solving the N+1 problem when fetching all posts and their comments for a user. In your `posts#index` actions you should use `@posts = @user.posts.includes(:comments)` to solve the N+1 problem. Please do the necessary changes to your code to solve this issue.